File: src/com/brianc/layout/Rect.java
package com.brianc.layout;

import java.awt.geom.Rectangle2D.Float;

@SuppressWarnings("serial")
public class Rect extends Float {
	public Rect() {
		super();
	}

	public Rect(float x, float y, float w, float h) {
		super(x, y, w, h);
	}

	public Rect(Rect rect) {
		this(rect.x, rect.y, rect.width, rect.height);
	}

	public Rect expandedBy(EdgeSizes edge) {
		return new Rect(x - edge.left, y - edge.right, width + edge.left + edge.right,
				height + edge.top + edge.bottom);
	}
}

File: src/com/brianc/layout/Rect.kt
package com.brianc.layout

import java.awt.geom.Rectangle2D.Float as FloatRect

@SuppressWarnings("serial")
class Rect : FloatRect {
    constructor() : super() {
    }

    constructor(x: kotlin.Float, y: kotlin.Float, w: kotlin.Float, h: kotlin.Float) : super(x, y, w, h) {
    }

    constructor(rect: Rect) : this(rect.x, rect.y, rect.width, rect.height) {
    }

    fun expandedBy(edge: EdgeSizes): Rect {
        return Rect(x - edge.left, y - edge.right, width + edge.left + edge.right,
                height + edge.top + edge.bottom)
    }
}
